datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -1414,4 +1398,23 @@ mod tests {
 
         Ok(())
     }
+
+    /// Create vector batches
+    fn create_vec_batches(schema: &Schema, n: usize) -> Vec<RecordBatch> {
+        let batch = create_batch(schema);
+        let mut vec = Vec::with_capacity(n);
+        for _ in 0..n {
+            vec.push(batch.clone());
+        }
+        vec
```

Comment:
```suggestion
        (0..n).map(|_| batch.clone()).collect()
```
And in the various copies of this function. It might even be sufficiently concise to obviate the need for duplicating the function